cou/apps/auxiliary.py

```diff
@@ -0,0 +1,180 @@
+
+        :param target: OpenStack release as target to upgrade.
+        :type target: OpenStackRelease
+        :return: The next channel for the application. E.g: victoria/stable
+        :rtype: str
+        """
+        return f"{self.openstack_to_track(target)}/stable"
+
+    def _get_refresh_charm_plan(
+        self, target: OpenStackRelease, parallel: bool = False
+    ) -> Optional[UpgradeStep]:
+        """Get plan for refreshing the current channel.
+
+        This function also identifies if charm comes from charmstore and in that case,
+        makes the migration.
+        :param target: OpenStack release as target to upgrade.
+        :type target: OpenStackRelease
+        :param parallel: Parallel running, defaults to False
+        :type parallel: bool, optional
+        :return: Plan for refreshing the charm.
+        :rtype: Optional[UpgradeStep]
+        """
+        switch = None
+        channel = self.channel
+
+        description = f"Refresh '{self.name}' to the latest revision of " f"'{self.channel}'"
+
+        if self.charm_origin == "cs":
+            description = f"Migration of '{self.name}' from charmstore to charmhub"
+            switch = f"ch:{self.charm}"
+            channel = self.expected_current_channel
+        elif self.channel != self.expected_current_channel:
+            logger.warning(
+                "'%s' has the channel set to: %s that is different from the expected channel: %s",
+                self.name,
+                self.channel,
+                self.expected_current_channel,
+            )
+
+        return UpgradeStep(
+            description=description,
+            parallel=parallel,
+            function=async_upgrade_charm,
+            application_name=self.name,
+            channel=channel,
+            model_name=self.model_name,
+            switch=switch,
+        )
+
+    def upgrade_plan(self, target: OpenStackRelease) -> list[Optional[UpgradeStep]]:
+        """Upgrade planning.
+
+        Auxiliary charms have multiple compatible OpenStack releases. In that case,
+        we check also the OpenStack origin from the "source" or "openstack-origin" to know
+        if it's necessary to change it. E.g: rabbitmq-server with workload version 3.8 is
+        compatible from ussuri to yoga. Even that is considered as yoga, we need to set the
+        source accordingly with the OpenStack components that might be in a lower version than
+        yoga.
+
+        :param target: OpenStack release as target to upgrade.
+        :type target: OpenStackRelease
+        :raises HaltUpgradePlanGeneration: When the application halt the upgrade plan generation
+        :return: Plan that will add upgrade as sub steps.
+        :rtype: list[Optional[UpgradeStep]]
+        """
+        os_origin_config = self.os_origin_config(target)
+        if self.current_os_release >= target and os_origin_config and os_origin_config >= target:
```

Comment:
If I', correct we can can compare `None >= target` or not? If yes the `and os_origin_config` is not needed.

Reply:
It won't throw an exception, but comparing None with target will always result to false. I think we can remove `and os_origin_config`